telemetry/common.go

```diff
@@ -0,0 +1,51 @@
+package telemetry
+
+import (
+	cosmostelemetry "github.com/cosmos/cosmos-sdk/telemetry"
+	"github.com/hashicorp/go-metrics"
+)
+
+// MetricNameKeys constructs the full metric name by prefixing with a defined
+// prefix and appending any additional metrics provided as variadic arguments.
+func MetricNameKeys(metrics ...string) []string {
+	result := make([]string, 0, len(metrics)+1)
+	result = append(result, metricNamePrefix)
+	result = append(result, metrics...)
+	return result
+}
+
+// isTelemetyEnabled returns whether is telemetry turned on in the config file `app.toml` - cosmos-sdk's telemetry section.
+func isTelemetyEnabled() bool {
+	return cosmostelemetry.IsTelemetryEnabled()
+}
+
+// appendMediumCardinalityLabels only creates the label if cardinality if set to "medium".
+// Good example of a medium cardinality label is `service_id` — we do not control the number of services
+// on the network, and as permissionless services grow the metrics can get easily out of hand. We're keeping
+// an option to turn off such labels.
+// Medium cardinality labels are included when the cardinality is set to "high".
```

Comment:
```suggestion
// A good example for a "medium" cardinality use-case is `service_id`:
// 	- This is a network wide parameter
//      - It is dependenon the permissionless nature of the network and can grow unbounded
//      - We're keeping an option to turn off such labels to avoid metric bloat
```